gptresearcher/gpt_researcher.py

```diff
@@ -0,0 +1,109 @@
+from typing import Optional
+import os
+
+import openai
+
+from gptresearcher.context.research_context import ResearchContext
+from gptresearcher.utils.setup_check import check_agent_setup, check_openai_api_key
+
+
+class GPTResearcher:
```

Comment:
@gregdrizz please see how to pass config file string as alternative to all these params. Consider rewriting our config file to be a JSON structure that can be imported into the class. Also consider modifying all these params to use **kwargs for a cleaner use. Please see examples for such best practices in other open sources or in general.

See here very broad example:
class MyClass(object):
    def __init__(self, **kwargs):
        for key, value in kwargs.items():
            setattr(self, key, value)

    @classmethod
    def from_json(cls, file_path):
        with open(file_path, 'r') as f:
            config = json.load(f)
        return cls(**config)
        
obj1 = MyClass(param1=value1, param2=value2, ...)
obj2 = MyClass.from_json('path/to/config.json')

Please make sure that optional params (such as debug_mode, user_agent, etc), I can think of 80% of these params being optional, that they are considered in **kwargs and dealt with a default value in __init__.